Dockerfile

```diff
@@ -65,20 +65,22 @@ RUN set -ex && cd ~ \
   && rm -vf /tmp/requirements.txt
 
 # apt-get all the things
+# Notes:
+# - Add all apt sources first
+# - Install gawk so the terraform-docs pre-commit hack works
 ARG CACHE_APT
 RUN set -ex && cd ~ \
-  && apt-get -qq update \
```

Comment:
It's been awhile since I've been on a distro that uses `apt`. Doesn't `update` also update the package index, meaning moving this to the end would cause package versions to be installed from the index in the base image (rather than the most up to date versions)? Explainshell gave me this:

> $ explain apt-get update
>
>
> apt-get(8) update
>
>APT package handling utility - - command-line interface
>
>update
>    update is used to resynchronize the package index files from their sources. The indexes of available
>    packages are fetched from the location(s) specified in /etc/apt/sources.list. For example, when using
>    a Debian archive, this command retrieves and scans the Packages.gz files, so that information about
>    new and updated packages is available. An update should always be performed before an upgrade or
>    dist-upgrade. Please be aware that the overall progress meter will be incorrect as the size of the
 >   package files cannot be known in advance.

Reply:
Ah I see, the update is moved to after all the manual steps involving curl but it's still in front of the `apt-get install`. So this is fine.